### PR TITLE
Issue #1410: JSON export should not output empty "comments" for candidat...

### PR DESCRIPTION
**Pull request by mjshepherd**
...e genes

### CODE REVIEW

File: components/patient-data/impl/src/main/java/org/phenotips/data/internal/controller/RejectedGeneListController.java [hunk 1]
```diff
@@ -142,8 +144,12 @@ public void writeJSON(Patient patient, JSONObject json, Collection<String> selec
                 json.put(getJsonPropertyName(), new JSONArray());
                 container = json.getJSONArray(getJsonPropertyName());
             }
+
+            if (item != null && "".equals(item.get(COMMENTS_KEY))) {
+                item.remove(COMMENTS_KEY);
+            }
+            container.add(item);
             container.add(item);
```
Comment (antonkatz) [inline components/patient-data/impl/src/main/java/org/phenotips/data/internal/controller/RejectedGeneListController.java]: Why are there two ```container.add(item)```?

File: components/patient-data/impl/src/main/java/org/phenotips/data/internal/controller/RejectedGeneListController.java [hunk 2]
```diff
@@ -142,8 +144,12 @@ public void writeJSON(Patient patient, JSONObject json, Collection<String> selec
                 json.put(getJsonPropertyName(), new JSONArray());
                 container = json.getJSONArray(getJsonPropertyName());
             }
+
+            if (item != null && "".equals(item.get(COMMENTS_KEY))) {
```
Comment (sdumitriu) [inline components/patient-data/impl/src/main/java/org/phenotips/data/internal/controller/RejectedGeneListController.java]: Use this:

    if (item != null && StringUtils.isBlank(item.get(COMMENTS_KEY))) {

File: components/patient-data/impl/src/main/java/org/phenotips/data/internal/controller/RejectedGeneListController.java [hunk 3]
```diff
@@ -41,6 +41,8 @@
 import javax.inject.Named;
 import javax.inject.Singleton;
 
+import org.apache.commons.lang.StringUtils;
```
Comment (sdumitriu) [inline components/patient-data/impl/src/main/java/org/phenotips/data/internal/controller/RejectedGeneListController.java]: Wrong import, should be `org.apache.commons.lang3.StringUtils` (notice the `3` in there).

File: components/patient-data/impl/src/main/java/org/phenotips/data/internal/controller/GeneListController.java [hunk 4]
```diff
@@ -41,6 +41,8 @@
 import javax.inject.Named;
 import javax.inject.Singleton;
 
+import org.apache.commons.lang.StringUtils;
```
Comment (sdumitriu) [inline components/patient-data/impl/src/main/java/org/phenotips/data/internal/controller/GeneListController.java]: Forgot this one.

**Pull request merged**